File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    ll = len(s)
    cnt = 0
    table = {"]": "[", "}": "{", ")": "("}
    s = s + s
    for i in range(ll):
        ss = s[i:i+ll]
        stack = []
        for j in range(ll):
            if ss[j] in {"[", "{", "("}:
                stack.append(ss[j])
            else:
                if stack and stack[-1] == table[ss[j]]:
                    stack.pop()
                else:
                    break
        else:
            if not stack:
                cnt += 1
    return cnt
    

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
const map = {
    '}': '{',
    ']': '[',
    ')': '(',
}

const openS = new Set(['{', '[', '('])

function solution(s) {
    function check(str) {
        let result = true;
        const stack = [];

        for (let nowS of str) {
            const isOpen = openS.has(nowS);

            if (isOpen) {
                stack.push(nowS);
                continue;
            }

            const stackS = stack.pop();

            if (map[nowS] !== stackS) {
                result = false;
                break;
            }
        }

        if (stack.length > 0) {
            return false
        }

        return result;
    }

    let result = 0

    for (let i=0; i<s.length; i++) {
        if (check(s.slice(i) + s.slice(0,i))) {
            result += 1
        }
    }

    return result;
}